src/scalar.rs

```diff
@@ -459,6 +462,28 @@ impl Default for Scalar {
     }
 }
 
+impl TryFrom<&[u8]> for Scalar {
+    type Error = CurveError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        if bytes.len() != 32 {
+            return Err(CurveError(
+                InternalError::BytesLengthError{name: "Scalar", length: 32}));
+        }
+
+        let mut s_bytes = [0x00u8; 32];
+        s_bytes.copy_from_slice(bytes);
+
+        Ok(Scalar::from_bits(s_bytes))
+    }
+}
+
+impl From<[u8; 32]> for Scalar {
+    fn from(bytes: [u8; 32]) -> Self {
+        Scalar{ bytes }
+    }
+}
+
```

Comment:
This would allow users to construct noncanonical `Scalar`s, both in that the most significant bit could be set, which is explicitly disallowed, and in that they could be unreduced modulo the basepoint order.